util/indexer/src/service.rs

```diff
@@ -238,13 +238,9 @@ impl IndexerService {
         self.async_handle.spawn(async move {
             let _initial_finished = initial_syncing.await;
             info!("initial_syncing finished");
-
-            tokio::select! {
-                _ = stop.cancelled() => {
-                    info!("Indexer received exit signal, cancel new_block_watcher task, exit now");
-                    return;
-                },
-                else => {},
+            if stop.is_cancelled() {
+                info!("Indexer received exit signal, cancel new_block_watcher task, exit now");
```

Comment:
if use rich-indexer,this content need update? auto get indexer type which is using

Reply:
Not needed. The version of CKB that includes rich-indexer has been refactored, and this check is moved to the module `indexer-sync`, which is common to both indexer and rich-indexer.